### PR TITLE
feat: bump KubeVirt to 1.0.0-150500.8.3.1

### PR DESCRIPTION
**Pull request by FrankYang0529**
SLE 15 SP4 general ends on 31 Dec 2023. KubeVirt base image will change. We have to update to a new version.

The supported version for SLE 15 SP5 will be 1.0.0: registry.suse.com/suse/sles/15.5/virt-operator:1.0.0-150500.8.3.1

**Related Issue:**
https://github.com/harvester/harvester/issues/4758

**Test plan:**

1. Test create/backup/snapshot/restore can work.
2. Test upgrade.

### CODE REVIEW

**Comment by FrankYang0529** [comment]
@bk201 I can create a new cluster and do some basic operations without error. However, I'm not sure why UI always show `VirtualMachine generation is 2, but latest observed generation is 1`. Could you help to take a look? Thank you.

Screenshot 2023-11-28 at 5 09 30 PM

**Comment by bk201** [comment]
@torchiaf We are bumping Kubevirt dependency, can you help check what's the condition to show those text? Thanks!

> @bk201 I can create a new cluster and do some basic operations without error. However, I'm not sure why UI always show `VirtualMachine generation is 2, but latest observed generation is 1`. Could you help to take a look? Thank you.
>

File: go.mod [hunk 1]
```diff
@@ -7,85 +7,75 @@ replace (
...
+	github.com/gobuffalo/flect v1.0.2
 	github.com/gorilla/mux v1.8.0
 	github.com/guonaihong/gout v0.1.3
 	github.com/harvester/harvester-network-controller v0.3.1
 	github.com/harvester/node-manager v0.1.5-0.20230614075852-de2da3ef3aca
 	github.com/iancoleman/strcase v0.2.0
 	github.com/k3s-io/helm-controller v0.11.7
-	github.com/k8snetworkplumbingwg/network-attachment-definition-client v0.0.0-20200331171230-d50e42f2b669
+	github.com/k8snetworkplumbingwg/network-attachment-definition-client v1.3.0
+	github.com/kube-logging/logging-operator/pkg/sdk v0.9.1
```
Comment (FrankYang0529) [inline go.mod:66]: @starbops It looks like `banzaicloud/logging-operator` has been changed to `kube-logging/logging-oeprator`. Could you check whether it's safe to do this change in `go.mod`? Thank you.

There is a function `loggingWorkaround`. Not sure whether we need do any update for it?

https://github.com/harvester/harvester/blob/acb6e69121a9843c8e7616de9f44401293c4d238/pkg/codegen/main.go#L200-L214

**Comment by FrankYang0529** [comment]
Upgrading from `v1.2.1` to this branch also works in my cluster.


Screenshot 2023-11-29 at 3 49 33 PM

File: go.mod (hunk 1 above)
Comment (starbops) [reply go.mod:66]: The GroupVersion has not changed, so keeping it untouched is okay. Thank you.
https://github.com/kube-logging/logging-operator/blob/bc3a4727c2dee2a362e85277a76ccff061e5e804/pkg/sdk/logging/api/v1beta1/groupversion_info.go#L27

**Comment by torchiaf** [comment]
> @torchiaf We are bumping Kubevirt dependency, can you help check what's the condition to show those text? Thanks!
> 
> > @bk201 I can create a new cluster and do some basic operations without error. However, I'm not sure why UI always show `VirtualMachine generation is 2, but latest observed generation is 1`. Could you help to take a look? Thank you.

I'm checking.

**Comment by torchiaf** [comment]
@FrankYang0529 `VirtualMachine generation is 2, but latest observed generation is 1` is the state description of the Object in the Resources table, in this case, the Virtual Machine.
The source is the field `metadata.state.message` from the response of `/v1/harvester/kubevirt.io.virtualmachines` endpoint.
Here is an example, (`clusters` in this case):

image

**Comment by bk201** [comment]
Rebased and remove non-changed charts in `deploy/charts/harvester/charts`

**Comment by ibrokethecloud** [comment]
the UI error is being caused due to the additional fields `desiredGeneration` and `obeservedGeneration` in new version of kubevirt. 

image


https://github.com/kubevirt/api/blob/v1.0.0/core/v1/types.go#L1527-L1537

Unless we decide to ignore it in the dashboard, this can lead to a bad UX with each VM reporting this error.

**Comment by ibrokethecloud** [comment]
Restarting the VM fixes the UI error.

The issue is being caused by vmi controller, which waits for VMI to be running, and then copies the interface mac addresses into the parent VM spec.

https://github.com/harvester/harvester/blob/master/pkg/controller/master/virtualmachine/vmi_network_controller.go#L19

Error will re-appear after a restart if an additional interface is added to the VM.

image

**Review by ibrokethecloud — APPROVED** [review]
lgtm. thanks.

**Review by bk201 — APPROVED** [review]
Thanks!

**Comment by FrankYang0529** [comment]
@Mergify backport v1.2

**Pull request merged**